sdk/core/Azure.Core/src/Pipeline/HttpPipelineMethod.cs

```diff
@@ -1,15 +1,100 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using System;
+
 namespace Azure.Core.Pipeline
 {
-    public enum HttpPipelineMethod
+    public readonly struct RequestMethod : IEquatable<RequestMethod>
     {
-        Get,
-        Post,
-        Put,
-        Patch,
-        Delete,
-        Head
+        public string Method { get; }
+
+        public static RequestMethod Get { get; } = new RequestMethod("GET");
+        public static RequestMethod Post { get; } = new RequestMethod("POST");
+        public static RequestMethod Put { get; } = new RequestMethod("PUT");
+        public static RequestMethod Patch { get; } = new RequestMethod("PATCH");
+        public static RequestMethod Delete { get; } = new RequestMethod("DELETE");
+        public static RequestMethod Head { get; } = new RequestMethod("HEAD");
+
+        public RequestMethod(string method)
+        {
+            Method = method.ToUpperInvariant();
+        }
+
+        public static RequestMethod Parse(string method)
+        {
+            if (method == null)
+            {
+                throw new ArgumentNullException(nameof(method));
+            }
+
+            // Fast-path common values
+            if (method.Length == 3)
+            {
+                if (string.Equals(method, "GET", StringComparison.OrdinalIgnoreCase))
+                {
+                    return Get;
+                }
+
+                if (string.Equals(method, "PUT", StringComparison.OrdinalIgnoreCase))
+                {
+                    return Put;
+                }
+            }
+            else if (method.Length == 4)
+            {
+                if (string.Equals(method, "POST", StringComparison.OrdinalIgnoreCase))
+                {
+                    return Post;
+                }
+                if (string.Equals(method, "HEAD", StringComparison.OrdinalIgnoreCase))
+                {
+                    return Head;
+                }
+            }
+            else
+            {
+                if (string.Equals(method, "PATCH", StringComparison.OrdinalIgnoreCase))
+                {
+                    return Patch;
+                }
+                if (string.Equals(method, "DELETE", StringComparison.OrdinalIgnoreCase))
+                {
+                    return Delete;
+                }
+            }
+
+            throw new ArgumentException($"'{method}' is not a known HTTP method");
+        }
+
+        public bool Equals(RequestMethod other)
+        {
+            return string.Equals(Method, other.Method, StringComparison.Ordinal);
+        }
+
+        public override bool Equals(object obj)
+        {
+            return obj is RequestMethod other && Equals(other);
+        }
+
+        public override int GetHashCode()
+        {
+            return Method.GetHashCode();
```

Comment:
Is Method null here if someone doesn't call the ctor for the struct?

Reply:
Yep, fixing.